### PR TITLE
Fix issues with payments and receipt items

### PR DESCRIPTION
**Pull request by kitsuta**
Fixes https://jira.magfest.net/browse/MAGDEV-613 and fixes self-service refunds. Fixes https://jira.magfest.net/browse/MAGDEV-595. Fixes https://jira.magfest.net/browse/MAGDEV-576.

### CODE REVIEW

File: uber/site_sections/preregistration.py [hunk 1]
```diff
@@ -807,15 +806,17 @@ def abandon_badge(self, session, id):
                                        failure_message)
                 elif response:
                     amount_refunded += response.amount
+                    for receipt_item in [item for item in attendee.receipt_items if item.txn_type == c.PAYMENT]:
```
Comment (EliAndrewC) [inline uber/site_sections/preregistration.py]: When you originally implemented the ``txn_type`` field, it made sense to me, but for some reason my brain now always expands it to "taxation type".

File: uber/utils.py [hunk 2]
```diff
@@ -1060,10 +1060,10 @@ def charge_cc(self, session, token):
         else:
             if self.models:
                 self._stripe_transaction = self.stripe_transaction_from_charge()
+                session.add(self._stripe_transaction)
                 for model in self.models:
                     multi = len(self.models) > 1
                     session.add(self.stripe_transaction_for_model(model, self._stripe_transaction, multi))
-                session.add(self._stripe_transaction)
```
Comment (EliAndrewC) [inline uber/utils.py]: Out of curiosity, why was this moved?  It's not obvious to me why it would make a difference.

Comment (kitsuta) [reply uber/utils.py]: Oh, I at first had no idea what was happening with the session, so I moved this just to see if it changed anything. I don't think it does, though.
